.github/workflows/test.yml

```diff
@@ -15,13 +15,15 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-24.04, macos-14, windows-2022]
+        os: [ubuntu-24.04, macos-14, windows-2022, macos-14-large]
         feature-args: ['', '-Funstable-mobile-app']
         include:
           - os: ubuntu-24.04
             display-os: Linux
           - os: macos-14
             display-os: macOS
+          - os: macos-14-large
+            display-os: macOS (Intel)
```

Comment:
As discussed in Slack, I would rather we test the Intel binary on the `macos-14` runner to avoid depending on additional runner images.

I suggest doing this like so: https://github.com/getsentry/sentry-cli/pull/2658 (this PR also adds linting for the Intel binary, which I think is also important to do).